Sourcery/Generating/Template/Swift/SwiftTemplate.swift

```diff
@@ -157,11 +164,85 @@ class SwiftTemplate: Template {
             throw SwiftTemplateParsingError.compilationError(path: binaryFile, error: compilationResult.error)
         }
 
-        let result = try Process.runCommand(path: binaryFile.description,
-                                            arguments: [serializedContextPath.description])
+        return binaryFile
+    }
+
+}
+
+class SwiftTemplate: NSObject, NSCoding, Template {
+
+    var sourcePath: Path
+    let code: String
+    let contentSha: String?
+
+    private lazy var buildDir: Path = Path.cleanTemporaryDir(name: "build")
 
+    private(set) var binaryPath: Path!
+
+    init(code: String, sourcePath: Path) {
+        self.code = code
+        self.sourcePath = sourcePath
+        self.contentSha = code.sha256()
+    }
+
+    init(path: Path, cachePath: Path?) throws {
```

Comment:
This could be transformed into convenience init that only takes `path`.  It seems to me that `cachePath` is only related with `loadOrBuild` method, but I've given a suggestion to transform it into an extension on SwiftTemplate.

```
func build(buildDir: Path, cachePath: Path?) throws -> Path {
```

This would separate `SwiftTemplate` which could be considered as a value type with only `let code: String`.

I think it would make sense to extend `func render(types: Types, arguments: [String: NSObject])` with `cachePath` so we could use `build` inside of it.

Alternative would be to also store `let cachePath: String?` and `let source: String` in `SwiftTemplate`, but I would still call `build` method inside `render` and make this just a dummy value type, although it's a `NSObject`.